parsec/core/gui/main_window.py

```diff
@@ -553,19 +562,21 @@ def add_instance(self, start_arg: Optional[str] = None):
             except ValueError as exc:
                 show_error(self, _("TEXT_INVALID_URL"), exception=exc)
 
-        idx = self._get_login_tab_index()
-        if idx != -1:
-            self.switch_to_tab(idx)
-        else:
-            tab = self.add_new_tab()
-            tab.show_login_widget()
-            self.on_tab_state_changed(tab, "login")
-            self.switch_to_tab(self.tab_center.count() - 1)
-            idx = self.tab_center.count() - 1
+        # Do not open a new logging widget if the organisation is already opened
+        if (
+            action_addr
+            and isinstance(action_addr, BackendOrganizationFileLinkAddr)
+            and self.tab_center.count()
+        ):
+            self.go_to_file_link(action_addr)
+            return
+
+        self.switch_to_login_tab()
 
         self.show_top()
         if action_addr and isinstance(action_addr, BackendOrganizationFileLinkAddr):
```

Comment:
I seems weird to have this check which is really similar to the one above, do you think we could factorize this ?
typically to end up with something like:
```python
if not action_addr:
    self.show_login_widget()  # Display login, creating the widget if neeeded
elif isinstance(action_addr, BackendOrganizationFileLinkAddr):
    self.go_to_file_link(action_addr)  # Find the wiget logged in with the organization or fallback to `self.show_login_widget(available_device_for_the_needed_organization, start_arg)` with an additional dialog
elif isintance(action_addr, BackendOrganizationBootstrapAddr):
    self.show_create_org_widget(action_addr)
elif isintsance(..):  # Similar stuff for device/user invitation urls
...
else:
    show_error(self, _("TEXT_INVALID_URL"))
```

Reply:
That's better, i refactored it to follow this logic (i also updated unit test to cover all kind of link)